queries/julia/highlights.scm

```diff
@@ -33,6 +33,103 @@
   (identifier) @function.call
   (#any-of? @_pipe "|>" ".|>"))
 
+; Function definitions
+; function f()
+(function_definition
+  (signature
+    (call_expression
+      [
+        (identifier) @function
+        (field_expression
+          (identifier) @function .)
+      ])))
```

Comment:
This inner query is shared among all of them but I couldn't find a way to collapse the queries to not repeat this snippet all the time...

Reply:
Is this not something that can be handled by more specific nodes in the parser?

Reply:
@savq any thoughts on this?

Reply:
Just checking, `(signature)` is only emitted by the parser for `(function_definition)` right?

> Also, I'd suggest not including patterns for where A where B, since that can be written as where {A, B}.

I mean, `f() = ...` can be written like `function f() ... end`. Just seems weird to tell people to rewrite valid code to get the correct highlighting. From my survey in the OP though double (or more) `where` is pretty rare.

Reply:
> Just checking, (signature) is only emitted by the parser for (function_definition) right?

Also for `macro_definition` 🙃 

>  From my survey in the OP though double (or more) where is pretty rare.

Yes, I was thinking mostly about the rarity of the pattern. Not that people should rewrite their code.